v3/sockjs/xhr.go

```diff
@@ -35,17 +35,18 @@ func (h *Handler) xhrSend(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	h.sessionsMux.Lock()
-	defer h.sessionsMux.Unlock()
-	if sess, ok := h.sessions[sessionID]; !ok {
+	sess, ok := h.sessions[sessionID]
```

Comment:
@fivegreenapples It looks as if the jsonp protocol has the same pattern -- locking the `sessionsMux` and then performing some additional (heavier) work before releasing the lock. Would you mind applying the same change, there?

@igm Any concerns with narrowing the mutex?

Reply:
I'll have a look over the weekend if I get time, otherwise next week. Thanks for the quick follow up.

Reply:
lgtm, I think we should do the same in `jsonp`.